pkg/controllers/provisioning/scheduling/scheduling_benchmark_test.go

```diff
@@ -162,10 +162,20 @@ func benchmarkScheduler(b *testing.B, instanceCount, podCount int) {
 		b.Fatalf("creating topology, %s", err)
 	}
 
-	scheduler := scheduling.NewScheduler(ctx, client, []*v1.NodePool{nodePool},
```

Comment:
Can we consider either adding a benchmark test in this PR or in a follow-on PR that validates what the performance of scheduling is with the addition of the reservation code? We should: 1) Compare our current scheduling performance and 2) Ensure that we add a test case for using reserved capacity type so that we can ensure that we are tracking when the performance improves over time